### DISCUSSION TITLE
What's the use of `tokenUrl` in OAuth2PasswordBearer

### DISCUSSION THREAD

**Discussion by r-priyam** [fastapi/fastapi#6602] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.security import OAuth2PasswordBearer

oauth =  OAuth2PasswordBearer(tokenUrl="...")
```

### Description

Hello there, Was just wondering that what is the use of `tokenUrl` in `OAuth2PasswordBearer(toeknUrl='')`. That's it :)

### Operating System

Windows

### FastAPI Version

0.68.1

### Python Version

3.9.7

**Comment by r-priyam** [comment]
Found the answer, My bad, I should lurk more before I ask 🤒

**Comment by ShinJam** [comment]
I haven't found it yet, can you tell me any references? What is it used for?

**Comment by JarroVGIT** [comment, marked as answer]
The `tokenUrl` parameter tells the `OAuth2PasswordBearer` logic where the token can be gotten from. This is primarily used for the `openapi.json` and the automatic docs. They will show API endpoints with the little lock, and when entering your user/password it will send a POST request with those user/password credentials to the `tokenUrl` endpoint to collect a bearer token.

Does that help you?

**Comment by ShinJam** [comment]
> Does that help you?

I see. Thank you for your answer It helped me a lot

**Comment by psanwal** [comment]
Thanks @JarroVGIT , been looking for this at so many places.
